airflow/utils/helpers.py

```diff
@@ -317,3 +318,26 @@ def render_log_filename(ti, try_number, filename_template):
                                     task_id=ti.task_id,
                                     execution_date=ti.execution_date.isoformat(),
                                     try_number=try_number)
+
+
+def dict_merge(dct, merge_dct):
```

Comment:
This doesn't cope with lists (other than taking the last value, maybe that's fine here?) - using https://github.com/toumorokoshi/deepmerge I think is slightly preferable - it has tests and is easy to configure which strategy we want.

(I've thought about adding this module in for use in a similar cas in the EmrCreateJobFlowOperator)

Reply:
That is so funny I want to use that for EmrCreateJobFlowOperator aswell!

Reply:
@ashb Not sure I fully understand how the policy works yet, can you point me to any examples?

Reply:
Somewhere in your python path (probably in `AIRFLOW_HOME`) you create an airflow_local_settings.py that contains:


```python
BASE_EXECUTOR_CONFIG = {
    # ...
}

def policy(task):
    # This fn will be called for every task of a dag as it gets "bagged"
    task.executor_config = dict_merge(BASE_EXECUTOR_CONFIG, getattr(task, 'executor_config', {})
```

Give or take. (Untested.)

Reply:
Having a dedicated/more direct way may be the way to do this, but I feel this policy feature is not obvious enough and the config file _may_ be big enough already.

I'm not against this PR, just pointing out ways it could be achieved right now.

Reply:
@ashb Makes sense now, thanks for explaining! I'm ok with either way if they achieve the same result. In an ideal world, I would love it if we could use `yaml` instead of the `airflow.cfg` file it would probably take a lot of work but I think it would be worth it for these kind of configuration situations.

If we go forward with this approach would you like it if I used that deepmerge package?

Reply:
> If we go forward with this approach would you like it if I used that deepmerge package?

Yes please.

I'm not personally a fan of YAML files for config -- I'm too prone to making whitespace errors and having a key in the wrong place :(